spec/rspec/matchers/dsl_spec.rb

```diff
@@ -29,6 +29,46 @@ def ok
     expect { matcher_b.i_dont_exist }.to raise_error(NameError)
   end
 
+  if RSpec::Support::RubyFeatures.required_kw_args_supported?
+    binding.eval(<<-CODE, __FILE__, __LINE__)
+    it 'supports the use of required keyword arguments in definition block' do
+      RSpec::Matchers.define(:match_required_kw) do |bar:|
+        match { expect(actual).to eq bar }
+      end
+      expect(1).to match_required_kw(bar: 1)
+    end
+
+    def kw(a:)
+      a
+    end
+
+    it "supports the use of required keyword arguments on methods" do
+      RSpec::Matchers.define(:matcher_required_kw_on_method) {}
+      expect(matcher_required_kw_on_method.kw(a: 1)).to eq(1)
+    end
+    CODE
+  end
+
+  if RSpec::Support::RubyFeatures.kw_args_supported?
+    binding.eval(<<-CODE, __FILE__, __LINE__)
+    it 'supports the use of optional keyword arguments in definition block' do
+      RSpec::Matchers.define(:match_optional_kw) do |bar: nil|
+        match { expect(actual).to eq bar }
+      end
+      expect(1).to match_optional_kw(bar: 1)
+    end
+
+    def optional_kw(a: nil)
+      a
+    end
+
+    it "supports the use of optional keyword arguments on methods" do
+      RSpec::Matchers.define(:matcher_optional_kw_on_method) {}
+      expect(matcher_optional_kw_on_method.optional_kw(a: 1)).to eq(1)
```

Comment:
I do not understand the link between `def optional_kw(a: nil)` and `matcher_optional_kw_on_method`. How `RSpec::Matchers.define(:matcher_optional_kw_on_method) {}` can use it?

Reply:
We pass `self` through to the matchers when they are created and they use `method_missing` to invoke the methods on example.

Reply:
ookk. I missed the `self`. Thanks